sql: Add empty pg_stat_activity virtual table

emsal0 opened the pull request:
Release note: Add an empty pg_stat_activity virtual table for
compatibility

closes #24745 

tested DBeaver and the client no longer errors out when trying to access Administer > Sessions

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/26249)
<!-- Reviewable:end -->

jordanlewis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

*[pkg/sql/pg_catalog.go, line 1905 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDmFBum8e8aTramFL9s:-LDmFBum8e8aTramFL9t:b-ukebrx) ([raw file](https://github.com/cockroachdb/cockroach/blob/a83a5daa25e670caae84f8327687dafdc4f4d433/pkg/sql/pg_catalog.go#L1905)):*
> ```Go
> 	username NAME,
> 	application_name TEXT,
> 	client_addr TEXT,
> ```

You should be able to use the INET type here, no?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-:-LDmFYKI5r0h85iUzz28:bnfp4nl)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
@jordanlewis ah right, I misread the spec.

knz commented:
Thanks for this quick PR. 

- update the release note in the commit message to use the correct format
- prefix the commit title with a scope e.g. `sql:`
- address the two code comments below

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/sql/logictest/testdata/logic_test/pg_catalog, line 1397 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDmGf0w95yKhL35tKVB:-LDmGf0x21cija-bokhS:b-546uhk) ([raw file](https://github.com/cockroachdb/cockroach/blob/a83a5daa25e670caae84f8327687dafdc4f4d433/pkg/sql/logictest/testdata/logic_test/pg_catalog#L1397)):*
> ```
> ----
> 
> query I
> ```

Instead make a query that reveals the column names and types. Follow the example of pg_catalog.pg_enum earlier in this file.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-:-LDmGlqr7zTkGLM2ALvq:b-ts544k)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[pkg/sql/logictest/testdata/logic_test/pg_catalog, line 1397 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDmGf0w95yKhL35tKVB:-LDmJfFn95dyCk5Vq0Gj:b-c1wzti) ([raw file](https://github.com/cockroachdb/cockroach/blob/a83a5daa25e670caae84f8327687dafdc4f4d433/pkg/sql/logictest/testdata/logic_test/pg_catalog#L1397)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Instead make a query that reveals the column names and types. Follow the example of pg_catalog.pg_enum earlier in this file.
</blockquote></details>

Not sure if it's showing up here, but I do have a test like that existing at line 134. Should I remove this test?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[pkg/sql/pg_catalog.go, line 1905 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDmFBum8e8aTramFL9s:-LDmOb2t2S68yoTMfazh:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a83a5daa25e670caae84f8327687dafdc4f4d433/pkg/sql/pg_catalog.go#L1905)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

@emsal1863: when you've addressed a comment, click the "done" or "Acknowledge" button in reviewable.
</blockquote></details>

Done.

---

*[pkg/sql/logictest/testdata/logic_test/pg_catalog, line 1397 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDmGf0w95yKhL35tKVB:-LDmQQwP7Hk5iknhy8Jr:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a83a5daa25e670caae84f8327687dafdc4f4d433/pkg/sql/logictest/testdata/logic_test/pg_catalog#L1397)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

This is a different test. What I am talking about is `SELECT * FROM pg_catalog.pg_stat_activity`
with the `colnames` option


</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249)*
<!-- Sent from Reviewable.io -->

knz commented:
Ok you can squash this and don't forget the commit title and the release note.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-:-LDmSwhy8rHhGAMxZS84:b-p60iha)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
Not sure what's going on in the TC failures because they appear to be failing in the same logic tests throughout multiple reruns. The logic tests are in areas other than `pg_catalog` (namely `default`, `opt`, and `distsql`).

benesch commented:
This time, they're not flakes! :)

For better or worse, we have logic tests that check the output of `select table_schema, table_name FROM information_schema.tables` and the like. They need to be updated whenever you add a new system table.

We have a script that automates this update:

```
make testlogic FLAGS=-rewrite-results-in-testfiles
```

But be very, very careful to review the resulting diff and make sure the changes are correct. You can also do it by hand; it's usually very obvious how to update the results appropriately.

BramGruneir commented:
I suggest doing the updates by hand!  Makes sure everything is correct that
way.

On Wed, May 30, 2018, 18:08 Nikhil Benesch <notifications@github.com> wrote:

> This time, they're not flakes! :)
>
> For better or worse, we have logic tests that check the output of select
> table_schema, table_name FROM information_schema.tables and the like.
> They need to be updated whenever you add a new system table.
>
> We have a script that automates this update:
>
> make testlogic FLAGS=-rewrite-results-in-testfiles
>
> But be very, very careful to review the resulting diff and make sure the
> changes are correct. You can also do it by hand; it's usually very obvious
> how to update the results appropriately.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/26249#issuecomment-393335802>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABihuTMk4KR9SBoLvHp12Qorb_f7Kz32ks5t3xhygaJpZM4UTv8_>
> .
>

emsal0 committed:
sql: Add the empty pg_stat_activity virtual table

Release note (sql change): Added an empty pg_stat_activity virtual table for
compatibility with DBeaver and other SQL clients that require it.

emsal1863 commented:
Everything seems to be in order; @knz are we good to go?

BramGruneir commented:
LGTM with a few comments to fix.

Also, can you add a link in the commit message to the table description in the postgres docs.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[pkg/sql/logictest/testdata/logic_test/information_schema, line 960 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDrpkRWCqRX1Rr0YyHf:-LDrpkRWCqRX1Rr0YyHg:b412ior) ([raw file](https://github.com/cockroachdb/cockroach/blob/1113c578d051d7ead4eed9314261fb9d28faea57/pkg/sql/logictest/testdata/logic_test/information_schema#L960)):*
> ```
> # root can see everything
> query TTTTTTTT colnames
> SELECT * FROM system.information_schema.table_privileges ORDER BY table_name
> ```

This seems to be changing the order around.  Can you add a more strict ordering to this to avoid it flaking?

I'm thinking, `table_name`, then `grantee`, then privilege type.

---

*[pkg/sql/logictest/testdata/planner_test/explain, line 719 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDrqMIkC-stugoc0By5:-LDrqMIkC-stugoc0By6:bsws13i) ([raw file](https://github.com/cockroachdb/cockroach/blob/29c01356fbe35810ab9ab54e566f9a4192505e5e/pkg/sql/logictest/testdata/planner_test/explain#L719)):*
> ```
>       │             render 0  (ARRAY[(1)[int],(2)[int],(3)[int]])[int[]]  ·             ·
>       └── emptyrow  ·         ·                                           ()            ·
> ```

Did you remove the extra empty line at the end of this file?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-:-LDroqRr-M5YpgNErDYv:bj3s8mu)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[pkg/sql/logictest/testdata/logic_test/information_schema, line 960 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDrpkRWCqRX1Rr0YyHf:-LDrzTRS7HApAQZuPPBx:bss0ksb) ([raw file](https://github.com/cockroachdb/cockroach/blob/1113c578d051d7ead4eed9314261fb9d28faea57/pkg/sql/logictest/testdata/logic_test/information_schema#L960)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

This seems to be changing the order around.  Can you add a more strict ordering to this to avoid it flaking?

I'm thinking, `table_name`, then `grantee`, then privilege type.
</blockquote></details>

I'll do exactly that.

---

*[pkg/sql/logictest/testdata/planner_test/explain, line 719 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-LDrqMIkC-stugoc0By5:-LDrzdoo7rDaEwenXPyd:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/29c01356fbe35810ab9ab54e566f9a4192505e5e/pkg/sql/logictest/testdata/planner_test/explain#L719)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

Did you remove the extra empty line at the end of this file?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249)*
<!-- Sent from Reviewable.io -->

knz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> modulo bram's comment. 

Another nit: make the text in the release notes full sentences, with preferably a capital at the beginning and in any case a period at the end.

Also things are not done "for compatibility", but instead "for compatibility *with* some other tech" (which needs to be named). Otherwise the reader doesn't know what to compare with to satisfy themselves that the change is doing what it advertises.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/26249#-:-LDrztWpAo1Huh6-jI9k:b-h2979j)*
<!-- Sent from Reviewable.io -->

emsal1863 commented:
@knz noted, will change the commit message accordingly when preparing for merge

jordanlewis commented:
LGTM

emsal1863 commented:
bors r+
